### PR TITLE
[Go] Add debug logging settings for common automation API operations

### PR DESCRIPTION
**Pull request by viveklak**

### CODE REVIEW

File: sdk/go/x/auto/example_test.go [hunk 1]
```diff
@@ -36,7 +35,7 @@ import (
 	"github.com/pulumi/pulumi/sdk/v2/go/x/auto/optup"
 )
 
-func Example() error {
+func Example() {
```
Comment (viveklak) [inline sdk/go/x/auto/example_test.go:38]: TIL that Godoc examples don't return errors.. My IDE was complaining about them but seems linter doesn't care? I can kill these changes if you prefer.

Comment (komalali) [reply sdk/go/x/auto/example_test.go:38]: I think this makes sense, handling errors will really be up to the user anyway 👍

**Review by komalali — COMMENTED** [review]
Is there any way we could test this?

**Comment by viveklak** [comment]
> Is there any way we could test this?

I have tested it with an automation api example myself but can't think of a nice way to add as an automated test. An e2e example would be hard to assert that debug logging is actually occurring. Could try and intercept the args passed to [`runPulumiCmdSync`](https://github.com/pulumi/pulumi/blob/9616bde745fbeaa032108bad7816be5b9a317c95/sdk/go/x/auto/stack.go#L734) but that also seems awkward.  Open to ideas on this...

**Review by komalali — APPROVED** [review]
@viveklak yeah that's fair, I can't think of a succinct automated test off the top of my head either.

**Comment by EvanBoyle** [comment]
Looks great! Sorry for being late on the review!

**Pull request merged**